### PR TITLE
[SPARK-34762][BUILD] Fix the build failure with Scala 2.13 which is related to commons-cli

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR fixes the build failure with Scala 2.13 which is related to `commons-cli`.
The last few days, build with Scala 2.13 on GA continues to fail and the error message says like as follows.
```
[error] /home/runner/work/spark/spark/sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java:26:1:  error: package org.apache.commons.cli does not exist
1278[error] import org.apache.commons.cli.GnuParser;
```
The reason is that `mvn help` in `change-scala-version.sh` downloads the POM file of `commons-cli` but doesn't download the JAR file, leading the build failure.

This PR also adds `commons-cli` to the dependencies explicitly because HiveThriftServer depends on it.
### Why are the changes needed?

Expect to fix the build failure with Scala 2.13.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

I confirmed that build successfully finishes with Scala 2.13 on my laptop.
```
find ~/.m2 -name commons-cli -exec rm -rf {} \;
find ~/.ivy2 -name commons-cli -exec rm -rf {} \;
find ~/.cache/ -name commons-cli -exec rm -rf {} \; // For Linux
find ~/Library/Caches -name commons-cli -exec rm -rf {} \; // For macOS

dev/change-scala-version 2.13
./build/sbt -Pyarn -Pmesos -Pkubernetes -Phive -Phive-thriftserver -Phadoop-cloud -Pkinesis-asl -Pdocker-integration-tests -Pkubernetes-integration-tests -Pspark-ganglia-lgpl -Pscala-2.13 clean compile test:compile
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Awesome @sarutak!

File: pom.xml [hunk 1]
```diff
@@ -2449,6 +2450,11 @@
           </exclusion>
         </exclusions>
       </dependency>
+      <dependency>
+        <groupId>commons-cli</groupId>
+        <artifactId>commons-cli</artifactId>
+        <version>1.4</version>
```
Comment (yaooqinn) [inline pom.xml:2456]: nit : ${commons-cli.version}

Comment (sarutak) [reply pom.xml:2456]: Oops. Thanks.

Comment (HyukjinKwon) [inline pom.xml]: BTW @sarutak, why is t 1.4? I just took a quick look and we pick it as 1.2: https://github.com/apache/spark/blob/master/dev/deps/spark-deps-hadoop-3.2-hive-2.3#L29

Comment (sarutak) [reply pom.xml]: Hmm, I just chose the latest version.
But it might be better to choose 1.2 for the behavior compatibility.
What do you think?

Comment (HyukjinKwon) [reply pom.xml]: Yeah, I think 1.2 is fine. We can upgrade it in another PR :-).

Comment (sarutak) [reply pom.xml]: O.K, let's choose `1.2` to focus on the issue for now.

File: pom.xml [hunk 2]
```diff
@@ -162,7 +162,7 @@
     <commons.math3.version>3.4.1</commons.math3.version>
     <!-- managed up from 3.2.1 for SPARK-11652 -->
     <commons.collections.version>3.2.2</commons.collections.version>
-    <scala.version>2.12.10</scala.version>
+    <scala.version>2.13.5</scala.version>
```
Comment (HyukjinKwon) [inline pom.xml:165]: I guess this is a mistake?

Comment (sarutak) [reply pom.xml]: Ah, Yes.It's related to #31863 (another issue).

**Review by HyukjinKwon — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]
great job!

**Comment by sarutak** [comment]
Actually, adding `commons-cli` explicitly doesn't resolve the issue but I noticed #31863 succeed to build with Scala 2.13. So I found, if `commons-cli` is already in `.m2`, the build will fail.
In `change-scala-version.sh`, `mvn` command runs and this downloads `commons-cli` to `.m2`.

So I added a workaround that removes `~/.m2/repository/commons-cli` in `change-scala-version.sh`.
Let's see the workaround works well.

After I found the workaround works, I'll change the title and description.

**Comment by sarutak** [comment]
I would found the real cause.
`build/mvn help:evaluate -Pscala-${TO_VERSION} -Dexpression=scala.version -q -DforceStdout` in `change-scala-version.sh` download **only** POM file and it's hash file for `commons-cli` and the jar file is not downloaded.

```
ls /home/kou/.m2/repository/commons-cli/commons-cli/1.2/
_remote.repositories  commons-cli-1.2.pom  commons-cli-1.2.pom.sha1
```

**Comment by sarutak** [comment]
Finally, build with Scala 2.13 seems to pass.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
I'm gonna merge this for other PRs.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.1.

**Comment by sarutak** [comment]
@HyukjinKwon 
Thanks for merging!
I was going to merge myself after I get agreement on the change but it's ok as I have approval from you now.

**Comment by dongjoon-hyun** [comment]
Thank you, @sarutak and all!

**Pull request closed without merging**